python/hail/utils.py

```diff
@@ -52,3 +56,140 @@ def functions_rst(self, file_name):
         Env.hail().utils.FunctionDocumentation.makeFunctionsDocs(file_name)
 
 
+@handle_py4j
+def hadoop_read(path, buffer_size=8192):
+    """Open a readable file through the Hadoop filesystem API. 
+    Supports distributed file systems like hdfs, gs, and s3.
+    
+    **Example:**
+    
+    .. doctest::
+        :options: +SKIP
+
+        >>> with hadoop_read('gs://my-bucket/notes.txt') as f:
+        ...     for line in f:
+        ...         print(line.strip())
+    
+    .. note::
+        
+        If Spark is running in cluster mode, both the source and destination 
+        file paths must be URIs (uniform resource identifiers). This means 
+        fully clarified paths, prefixed by scheme (``file://``, ``hdfs://``, ``gs://``,
+        ``s3://``, etc.)
```

Comment:
No, they don't need to be "fully clarified".  Hadoop interfaces work fine with normal paths, they'll just be interpreted relative to the default filesystem.